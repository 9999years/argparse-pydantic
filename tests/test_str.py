from argparse import ArgumentParser
import pytest
from pytest import CaptureFixture
from syrupy.assertion import SnapshotAssertion

from pydantic import Field
from pydantic_argparse_extensible import ArgModel


class Args(ArgModel):
    user: str = Field(description="""The username to connect to.""")

    host: str = Field(description="""The host to connect to.""")


def test_required(capsys: CaptureFixture, snapshot: SnapshotAssertion) -> None:
    with pytest.raises(SystemExit):
        Args.argparse(args=[])
    assert capsys.readouterr().err == snapshot


def test_default_argparser(capsys: CaptureFixture, snapshot: SnapshotAssertion) -> None:
    with pytest.raises(SystemExit):
        Args.argparse(parser=ArgumentParser(prog="puppy"), args=[])
    assert capsys.readouterr().err == snapshot


def test_parse() -> None:
    args = Args.argparse(args=["--user", "puppy", "--host", "doggy"])
    assert args == Args(user="puppy", host="doggy")
