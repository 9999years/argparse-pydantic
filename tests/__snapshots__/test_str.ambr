# serializer version: 1
# name: test_default_argparser
  '''
  usage: puppy [-h] --user USER --host HOST
  puppy: error: the following arguments are required: --user, --host
  
  '''
# ---
# name: test_required
  '''
  usage: pytest [-h] --user USER --host HOST
  pytest: error: the following arguments are required: --user, --host
  
  '''
# ---
